packages/backend/src/open_payments/payment/outgoing/service.ts

```diff
@@ -159,6 +185,181 @@ async function createOutgoingPayment(
   }
 }
 
+function validateAccess({
+  access,
+  payment,
+  identifier
+}: {
+  access: GrantAccess
+  payment: OutgoingPayment
+  identifier: string
+}): PaymentAccess | undefined {
+  let paymentInterval: Interval | undefined
+  if (access.interval)
+    paymentInterval = getInterval(access.interval, payment.createdAt)
+  if (
+    (!access.identifier || access.identifier === identifier) &&
+    (!access.interval || paymentInterval !== undefined) &&
+    (!access.limits || validateAccessLimits(payment, access.limits))
+  ) {
+    return { ...access, paymentInterval }
+  }
+}
+
+function validatePaymentAccess({
+  access,
+  payment,
+  identifier
+}: {
+  access: PaymentAccess
+  payment: OutgoingPayment
+  identifier: string
+}): boolean {
+  if (
+    (!access.identifier || access.identifier === identifier) &&
+    (!access.paymentInterval ||
+      (access.paymentInterval.start.toMillis() <= payment.createdAt.getTime() &&
+        payment.createdAt.getTime() < access.paymentInterval.end.toMillis())) &&
+    (!access.limits || validateAccessLimits(payment, access.limits))
```

Comment:
:thinking: If grants were immutable, we would only need to check if an existing payment is within the same interval.
Instead, we have the opportunity for an existing payment to no longer match `identifier`, `limits` fields, etc.

Reply:
Maybe we should just allow `limits` changes. Everything else requires a revocation of an old grant and the issuance of a new grant?
@adrianhopebailie @dclipp @njlie